File: Dot examples/ASP Book Examples/Book Examples/Chapter24/CS/App_Code/BinaryShoppingCart.cs
using System;
using System.Collections.Generic;
using System.Web.Profile;

namespace AspNetUnleashed
{
    [Serializable]
    public class BinaryShoppingCart
    {
        private List<BinaryCartItem> _items = new List<BinaryCartItem>();

        public List<BinaryCartItem> Items
        {
            get { return _items; }
        }
    }

    [Serializable]
    public class BinaryCartItem
    {
        private string _name;
        private decimal _price;
        private string _description;

        public string Name
        {
            get { return _name; }
            set { _name = value; }
        }

        public decimal Price
        {
            get { return _price; }
            set { _price = value; }
        }

        public string Description
        {
            get { return _description; }
            set { _description = value; }
        }

        public BinaryCartItem() { }

        public BinaryCartItem(string name, decimal price, string description)
        {
            _name = name;
            _price = price;
            _description = description;
        }
    }
}

File: Dot examples/ASP Book Examples/Book Examples/Chapter24/VB/App_Code/BinaryShoppingCart.vb
Imports System
Imports System.Collections.Generic
Imports System.Web.Profile

Namespace AspNetUnleashed

    <Serializable()> _
    Public Class BinaryShoppingCart
        Private _items As New List(Of BinaryCartItem)()

        Public ReadOnly Property Items() As List(Of BinaryCartItem)
            Get
                Return _items
            End Get
        End Property
    End Class

    <Serializable()> _
    Public Class BinaryCartItem
        Private _name As String
        Private _price As Decimal
        Private _description As String

        Public Property Name() As String
            Get
                Return _name
            End Get
            Set(ByVal Value As String)
                _name = Value
            End Set
        End Property

        Public Property Price() As Decimal
            Get
                Return _price
            End Get
            Set(ByVal Value As Decimal)
                _price = Value
            End Set
        End Property

        Public Property Description() As String
            Get
                Return _description
            End Get
            Set(ByVal Value As String)
                _description = Value
            End Set
        End Property

        Public Sub New()
        End Sub

        Public Sub New(ByVal name As String, ByVal price As Decimal, ByVal description As String)
            _name = name
            _price = price
            _description = description
        End Sub
    End Class
End Namespace
